app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,356 @@
+                .setNotify(ebeanExtension.getNotify())
+                .setMaxCheckAttempts(ebeanExtension.getMaxCheckAttempts())
+                .setFreshnessThresholdInSeconds(ebeanExtension.getFreshnessThreshold())
+                .build();
+    }
+
+    private NagiosExtension convertToEbeanNagiosExtension(final models.internal.NagiosExtension internalExtension) {
+        if (internalExtension == null) {
+            return null;
+        }
+        final NagiosExtension extension = new NagiosExtension();
+        extension.setSeverity(internalExtension.getSeverity());
+        extension.setNotify(internalExtension.getNotify());
+        extension.setMaxCheckAttempts(internalExtension.getMaxCheckAttempts());
+        extension.setFreshnessThreshold(internalExtension.getFreshnessThresholdInSeconds());
+        return extension;
+    }
+
+    private final AtomicBoolean _isOpen = new AtomicBoolean(false);
+    private final AlertQueryGenerator _alertQueryGenerator;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseAlertRepository.class);
+
+    /**
+     * Inteface for database query generation.
+     */
+    public interface AlertQueryGenerator {
+
+        /**
+         * Translate the <code>AlertQuery</code> to an Ebean <code>Query</code>.
+         *
+         * @param query The repository agnostic <code>AlertQuery</code>.
+         * @return The database specific <code>PagedList</code> query result.
+         */
+        PagedList<models.ebean.Alert> createAlertQuery(AlertQuery query);
+
+        /**
+         * Save the <code>Alert</code> to the database. This needs to be executed in a transaction.
+         *
+         * @param alert The <code>Alert</code> model instance to save.
+         */
+        void saveAlert(models.ebean.Alert alert);
+
+        /**
+         * Gets the etag for the alerts table.
+         *
+         * @return The etag for the table.
+         */
+        long getEtag();
+    }
+
+    /**
+     * RDBMS agnostic query for alerts.
+     */
+    public static final class GenericQueryGenerator implements AlertQueryGenerator {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public PagedList<models.ebean.Alert> createAlertQuery(final AlertQuery query) {
+            ExpressionList<models.ebean.Alert> ebeanExpressionList = Ebean.find(models.ebean.Alert.class).where();
+            if (query.getCluster().isPresent()) {
+                ebeanExpressionList = ebeanExpressionList.eq("cluster", query.getCluster().get());
+            }
+            if (query.getContext().isPresent()) {
+                ebeanExpressionList = ebeanExpressionList.eq("context", query.getContext().get().toString());
+            }
+            if (query.getService().isPresent()) {
+                ebeanExpressionList = ebeanExpressionList.eq("service", query.getService().get());
+            }
+
+            final Query<models.ebean.Alert> ebeanQuery = ebeanExpressionList.query();
```

Comment:
So the existing AlertQuery query supports a "contains" clause. I don't see support for that here. To prevent this from getting out of hand (you'll need to add a full text index and the corresponding query logic) please add an issue here and in Github and do the work as a fast-follow.